classes/helper.php

```diff
@@ -183,4 +183,48 @@ public static function full_request(\stdClass $profile): string {
         }
         return $displayedrequest;
     }
+
+    /**
+     * Make a single record for a histogram table.
+     * Row is of the form: 2^(k-1) - 2^k : 2^(v-1).
+     *
+     * @param int $durationexponent The exponent (k) of the high end of the duration range.
+     * @param int $count The fuzzy count (v), or zero if no value.
+     * @return array
+     */
+    private static function make_histogram_record(int $durationexponent, int $count): array {
+        $high = pow(2, $durationexponent);
+        $low = ($high == 1) ? 0 : pow(2, $durationexponent - 1);
+        $val = $count ? pow(2, $count - 1) : 0;
+        return [
+            'low'   => $low,
+            'high'  => $high,
+            'value' => $val,
+        ];
+    }
+
+    /**
+     * Create a histogram table for a page group.
+     *
+     * @param \stdClass $record Page group's record.
+     * @return array
+     */
+    public static function make_histogram(\stdClass $record): array {
+        $counts = json_decode($record->fuzzydurationcounts, true);
+        ksort($counts);
+
+        $histogramrecs = [];
```

Comment:
Could we use the full name here?

```suggestion
        $histogramrecords = [];
```

Reply:
done.